cask.el

```diff
@@ -806,6 +810,15 @@ url to the fetcher source."
         (setf (cask-dependency-ref dependency) ref))
       (-when-let (branch (plist-get args :branch))
         (setf (cask-dependency-branch dependency) branch)))
+    (-when-let (archive (plist-get args :archive))
+      (if (not (boundp 'package-pinned-packages))
+          (princ "Archive pinning is only supported with Emacs v24.4+.\n")
```

Comment:
Raise a proper error here. We do this all over the place in the code so take a look at how we do there.